src/shims/unix/foreign_items.rs

```diff
@@ -163,9 +163,23 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
             "ftruncate64" => {
                 let [fd, length] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let fd = this.read_scalar(fd)?.to_i32()?;
+                let length = this.read_scalar(length)?.to_i64()?;
                 let result = this.ftruncate64(fd, length)?;
                 this.write_scalar(result, dest)?;
             }
+            "ftruncate" => {
```

Comment:
If you're adding this for all targets now we should probably have a test for it... `libc-fs.rs` sound like the right file.

We've historically been pretty bad with libc tests for our FS APIs though. And `test_posix_mkstemp` should probably be in libc-fs.rs as well. So if you prefer we can open an issue to track fixing this in the future.

Reply:
There's still no test for this function, is there?

Reply:
I did not add purposely, can be done separately.

Reply:
I said it should either be done here or have an issue to track it. :)
Here's an issue: https://github.com/rust-lang/miri/issues/3179